kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,662 @@
+
+            this->mpScheme->InitializeNonLinIteration(r_model_part, r_A, r_Dx, r_b);
+            this->mpConvergenceCriteria->InitializeNonLinearIteration(r_model_part, r_dof_set, r_A, r_Dx, r_b);
+            is_converged = this->mpConvergenceCriteria->PreCriteria(r_model_part, r_dof_set, r_A, r_Dx, r_b);
+
+            TSparseSpace::SetToZero(r_A);
+            TSparseSpace::SetToZero(r_b);
+            TSparseSpace::SetToZero(r_Dxf);
+
+            // We compute r_Dxf 
+            this->mpBuilderAndSolver->Build(this->mpScheme, r_model_part, r_A, r_b);
+            this->mpBuilderAndSolver->ApplyDirichletConditions(this->mpScheme, r_model_part, r_A, r_Dxf, r_b);
+            TSparseSpace::Assign(r_b, 1.0, r_f);
+            this->mpBuilderAndSolver->SystemSolve(r_A, r_Dxf, r_b);
+
+            TSparseSpace::SetToZero(r_A);
+            TSparseSpace::SetToZero(r_b);
+            TSparseSpace::SetToZero(r_Dxb);
+
+            this->mpBuilderAndSolver->BuildAndSolve(this->mpScheme, r_model_part, r_A, r_Dxb, r_b);
+            lambda_increment = -TSparseSpace::Dot(r_DxPred, r_Dxb) / TSparseSpace::Dot(r_DxPred, r_Dxf);
+            TSparseSpace::Assign(r_Dx, 1.0, r_Dxb + lambda_increment*r_Dxf);
+
+            // Update results
+            mDLambdaStep += lambda_increment;
+            mLambda += lambda_increment;
+            TSparseSpace::UnaliasedAdd(r_DxStep, 1.0, r_Dx);
+            UpdateDatabase(r_A, r_Dx, r_b, BaseType::MoveMeshFlag());
+
+            this->mpScheme->FinalizeNonLinIteration(r_model_part, r_A, r_Dx, r_b);
+            this->mpConvergenceCriteria->FinalizeNonLinearIteration(r_model_part, r_dof_set, r_A, r_Dx, r_b);
+
+            if (is_converged) {
+                if (this->mpConvergenceCriteria->GetActualizeRHSflag()) {
+                    TSparseSpace::SetToZero(r_b);
+                    this->mpBuilderAndSolver->BuildRHS(this->mpScheme, r_model_part, r_b);
+                }
+                is_converged = this->mpConvergenceCriteria->PostCriteria(r_model_part, r_dof_set, r_A, r_Dx, r_b);
+            }
+        }
+        // Prints a warning if the maximum number of iterations is exceeded
+        if (iteration_number >= BaseType::mMaxIterationNumber) {
+            MaxIterationsExceeded();
+        } else {
+            KRATOS_INFO_IF("Arc-Length Strategy", this->GetEchoLevel() > 0)
+                << "Convergence achieved after " << iteration_number << " / "
+                << BaseType::mMaxIterationNumber << " iterations" << std::endl;
+        }
+        //calculate reactions if required
+        if (BaseType::mCalculateReactionsFlag)
+            this->mpBuilderAndSolver->CalculateReactions(this->mpScheme, r_model_part, r_A, r_Dx, r_b);
+        return is_converged;
+    }
+
+    /**
+     * @brief This method updates the value of the external load according to the new load factor Lambda
+     */
+    void UpdateExternalLoads()
```

Comment:
The only relevant missing suggestion, which I am not going to consider as blocking, and we can discuss in a further issue/PR is this part, which is not general enough, and too much hardcoded, and only works in some cases IMO.

The thing is that, IMO the easiest and most generic way to solve this is to consider a variable, for example LOAD_FACTOR, that is defined in the ProcessInfo, then in the loads, check for the existence of the definition of this variable, otherwise, consider 1. This will avoid resetting the values, and will work in all cases.

Reply:
I talked with @RiccardoRossi about this. He's not fully convinced. I any case we can have this conversation in another issue/PR :)